azure_data_cosmos_engine/src/query/mod.rs

```diff
@@ -87,6 +89,144 @@ impl PartitionKeyRange {
     }
 }
 
+#[derive(Clone, Debug, Deserialize)]
+// TODO: pk values here are all currently strings - we need the same sort of PartitionKeyValue
+// logic used in the main Rust SDK in order to compare and be able to use it within this method.
+pub struct ItemIdentity {
+    #[serde(rename = "PartitionKeyValue")]
+    partition_key_value: String,
+    #[serde(rename = "ID")]
+    id: String,
+}
+
+impl ItemIdentity {
+    pub fn new(id: impl Into<String>, partition_key_value: impl Into<String>) -> Self {
+        Self {
+            id: id.into(),
+            partition_key_value: partition_key_value.into(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct QueryChunk {
+    pub pk_range_id: String,
+    pub items: Vec<QueryChunkItem>
+}
+
+#[derive(Debug)]
+pub struct QueryChunkItem {
+    pub index: usize,
+    pub id: String,
+    pub partition_key_value: String,
+}
+
+impl QueryChunk {
+    pub fn from_identities(item_identities: Vec<ItemIdentity>, pkranges: &mut Vec<PartitionKeyRange>, pk_kind: PartitionKeyKind, pk_version: u8) -> Vec<QueryChunk> {
```

Comment:
```suggestion
    pub fn from_identities(item_identities: Vec<ItemIdentity>, pkranges: &mut Vec<PartitionKeyRange>, pk_kind: PartitionKeyKind, pk_version: u8) -> Vec<Self> {
```

Nit: This makes the diff smaller if you rename the type.